src/System.Numerics.Vectors/tests/GenericVectorTests.cs

```diff
@@ -2610,6 +2611,76 @@ public void NarrowDouble()
         #endregion Narrow / Widen
 
         #region Helper Methods
+        private static void AssertEqual<T>(T expected, T actual, string operation, int precision = -1) where T : IEquatable<T>
+        {
+            if (typeof(T) == typeof(float))
+            {
+                if (!IsDiffTolerable((float)(object)expected, (float)(object)actual, precision))
+                {
+                    throw new XunitException($"AssertEqual failed for operation {operation}. Expected: {expected}, Actual: {actual}.");
+                }
+            }
+            else if (typeof(T) == typeof(double))
+            {
+                if (!IsDiffTolerable((double)(object)expected, (double)(object)actual, precision))
+                {
+                    throw new XunitException($"AssertEqual failed for operation {operation}. Expected: {expected}, Actual: {actual}.");
+                }
+            }
+            else
+            {
+                if (!expected.Equals(actual))
+                {
+                    throw new XunitException($"AssertEqual failed for operation {operation}. Expected: {expected}, Actual: {actual}.");
+                }
+            }
+        }
+
+        private static bool IsDiffTolerable(double d1, double d2, int precision)
+        {
+            if (double.IsNaN(d1))
+            {
+                return double.IsNaN(d2);
+            }
+            if (double.IsInfinity(d1))
+            {
+                return AreSameInfinity(d1, d2 * 10);
+            }
+            if (double.IsInfinity(d2))
+            {
+                return AreSameInfinity(d1 * 10, d2);
+            }
+            double diffRatio = (d1 - d2) / d1;
+            diffRatio *= Math.Pow(10, precision);
+            return Math.Abs(diffRatio) < 1;
+        }
+
+        private static bool IsDiffTolerable(float f1, float f2, int precision)
+        {
+            if (float.IsNaN(f1))
+            {
+                return float.IsNaN(f2);
+            }
+            if (float.IsInfinity(f1))
+            {
+                return AreSameInfinity(f1, f2 * 10);
+            }
+            if (float.IsInfinity(f2))
+            {
+                return AreSameInfinity(f1 * 10, f2);
+            }
+            float diffRatio = (f1 - f2) / f1;
+            diffRatio *= (float)Math.Pow(10, precision);
```

Comment:
It might be worthwhile to create a MathF wrapper on NetFX (like I did here: https://github.com/dotnet/corefx/blob/master/src/System.Numerics.Vectors/src/System/MathF.cs).

It will catch any errors allowed by the additional precision of `Math.Pow(double)`.